### PR TITLE
Add example usage of appkit header middleware

### PR DESCRIPTION
**Pull request by sunfmin**

### CODE REVIEW

File: prottp.go [hunk 1]
```diff
@@ -58,6 +59,26 @@ func NewError(statusCode int, body proto.Message) ErrorWithStatus {
...
 }
 
+type key int
+
+const headerKey key = iota
+
+func WithHeader(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		newCtx := context.WithValue(ctx, headerKey, http.Header{})
```
Comment (bodhi) [inline prottp.go]: If you use `w.Header()` instead of `http.Header{}`, then there's no need to write headers in `WriteMessage`?

**Comment by sunfmin** [comment]
@bodhi Yes, you are write, Fixed.

**Comment by bodhi** [comment]
Then my next comment is:

Does this need to be in prottp? Can it just go as generic middleware in appkit?

**Comment by sunfmin** [comment]
I was going to ask about this... I'll move it to appkit.

**Comment by sunfmin** [comment]
@bodhi Which package in appkit? I saw middleware is scattered around in appkit packages.

**Comment by sunfmin** [comment]
@bodhi Moved!

**Comment by bodhi** [comment]
Oh, this wasn't a PR to merge into master...

**Pull request merged**